### PR TITLE
Added osu irc client class

### PR DESCRIPTION
**Pull request by Gr3g027**
The IRC works, whats left to do is picking up the osu link from a chat message and send it.

### CODE REVIEW

File: gosumemoryClient.cpp [hunk 1]
```diff
@@ -34,15 +34,15 @@ void GosumemoryClient::onConnected()
 void GosumemoryClient::onTextMessageReceived(QString message)
 {
 	if (readEnabled){
-		qDebug() << "GosumemoryClient: message received from: " << url.toString() << message;
+        // qDebug() << "GosumemoryClient: message received from: " << url.toString() << message;
```
Comment (Sclafus) [inline gosumemoryClient.cpp]: Watch out for whitespaces

File: gosumemoryClient.h [hunk 2]
```diff
@@ -28,7 +28,7 @@ class GosumemoryClient : public QObject {
 		void messageReceived(GosuMemoryDataWrapper message);
 		void disconnected();
 
-	private slots:
+    public slots:
```
Comment (Sclafus) [inline gosumemoryClient.h]: My bad, thanks for the correction
But watch out for whitespaces 👀

File: osuircclient.cpp [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+OsuIrcClient::OsuIrcClient(
+    const QString &nickname,
+    const QString &password,
+    const QString &server,
+    const int &port,
+    QObject *parent
+    ) : QObject(parent), nickname(nickname), password(password), server(server), port(port)
+{
+    // socket = new QTcpSocket(this);
+    connect(&socket, &QTcpSocket::connected, this, &OsuIrcClient::onConnected);
```
Comment (Sclafus) [inline osuircclient.cpp]: This fails the build
I don't know why

**Review by Sclafus — CHANGES_REQUESTED** [review]
Watch out for whitespaces, everything else looks good to me

File: osuircclient.cpp (hunk 3 above)
Comment (Gr3g027) [reply osuircclient.cpp]: It was the pointer variable socket, corrected

**Review by Sclafus — APPROVED** [review]
Looks good to me

**Pull request merged**